winsup/cygwin/uname.cc

```diff
@@ -37,9 +37,11 @@ uname_x (struct utsname *name)
       memset (name, 0, sizeof (*name));
       /* sysname */
       char* msystem = getenv("MSYSTEM");
-      const char *msystem_msys = "MSYS";
+      const char* msystem_sysname = "MSYS";
+      if (msystem != NULL && strlen(msystem) != 0 && strcmp(msystem, "MSYS") != 0)
```

Comment:
This is a more canonical test for a non-empty string:

```suggestion
      if (msystem != NULL && *msystem && strcmp(msystem, "MSYS") != 0)
```